pdata/pprofile/attributes.go

```diff
@@ -26,3 +30,35 @@ func FromAttributeIndices(table AttributeTableSlice, record attributable) pcommo
 
 	return m
 }
+
+// AddAttribute updates an AttributeTable and a record's AttributeIndices to
+// add a new attribute.
+// The record can by any struct that implements an `AttributeIndices` method.
+func AddAttribute(table AttributeTableSlice, record attributable, key string, value any) error {
+	for i := 0; i < table.Len(); i++ {
+		a := table.At(i)
+
+		if a.Key() == key && value == a.Value().AsRaw() {
+			if i >= math.MaxInt32 {
+				return fmt.Errorf("Attribute %s=%#v has too high an indice to be added to AttributeIndices", key, value)
+			}
+
+			record.AttributeIndices().Append(int32(i)) //nolint:gosec // overflow checked
+			return nil
+		}
+	}
+
+	if table.Len() >= math.MaxInt32 {
```

Comment:
Before this check ever triggers, many GB of RAM are used. Does it make sense to have a smaller limit, like 10k or 100k?

Reply:
This check is there only for the gosec linter when converting int into an int32.
While actually limiting the number of attributes may make sense, this is not the proper location to do it.

Reply:
Where else if not at the point where we append data?

Reply:
If we already have 100k of attributes here, it should have been limited by the receiver.

Reply:
More likely that processors are using this helper to add attributes.
Not a blocker, though.